rdrf/rdrf/views/form_view.py

```diff
@@ -1824,6 +1829,7 @@ def post(self, request, registry_code, patient_id, context_id=None):
             custom_consent_form.save()
             if patient_signature_form:
                 patient_signature_form.save()
+            self._move_to_next_stage(registry_model, patient_model)
```

Comment:
I don't think this is right.
1. The patient should be moved to `Eligibility` only if they have consented (ie. checked all the consent boxes)
2. Moving them to the next stage assumes that they are in the `Informed Consent` stage when the consent form is saved which isn't necessarily true.

I think we are looking towards having rules associated with stages that get evaluated when changes are made to a patient and decide whether the patient's stage should change.
In this case, the rules would be: current stage is 'Informed Consent' and all consent has been given.

At this stage of development, we could refrain going the full way, and just have a column with pre-defined rules on the `PatientStage`. ex. `PATIENT_REGISTERED`, `PATIENT_CONSENTED`.

Will see how things develop further down the line and whether we will have to give end-user more customisability by letting them provide their own rules.

Reply:
Updated.